src/compute/src/lib.rs

```diff
@@ -218,12 +224,238 @@ pub fn start(
+
+
+        NativeMethod {
+            name: JNIString::from("rowGetDecimalValue"),
+            sig: JNIString::from("(JI)Ljava/math/BigDecimal;"),
+            fn_ptr: risingwave_java_binding::Java_com_risingwave_java_binding_Binding_rowGetDecimalValue as *mut c_void,
+        },
+
+
+        NativeMethod {
+            name: JNIString::from("rowGetTimeValue"),
+            sig: JNIString::from("(JI)Ljava/sql/Time;"),
+            fn_ptr: risingwave_java_binding::Java_com_risingwave_java_binding_Binding_rowGetTimeValue as *mut c_void,
+        },
+
+
+        NativeMethod {
+            name: JNIString::from("rowGetDateValue"),
+            sig: JNIString::from("(JI)Ljava/sql/Date;"),
+            fn_ptr: risingwave_java_binding::Java_com_risingwave_java_binding_Binding_rowGetDateValue as *mut c_void,
+        },
+
+
+        NativeMethod {
+            name: JNIString::from("rowGetIntervalValue"),
+            sig: JNIString::from("(JI)Ljava/lang/String;"),
+            fn_ptr: risingwave_java_binding::Java_com_risingwave_java_binding_Binding_rowGetIntervalValue as *mut c_void,
+        },
+
+
+        NativeMethod {
+            name: JNIString::from("rowGetJsonbValue"),
+            sig: JNIString::from("(JI)Ljava/lang/String;"),
+            fn_ptr: risingwave_java_binding::Java_com_risingwave_java_binding_Binding_rowGetJsonbValue as *mut c_void,
+        },
+
+
+        NativeMethod {
+            name: JNIString::from("rowGetByteaValue"),
+            sig: JNIString::from("(JI)[B"),
+            fn_ptr: risingwave_java_binding::Java_com_risingwave_java_binding_Binding_rowGetByteaValue as *mut c_void,
+        },
+
+
+        NativeMethod {
+            name: JNIString::from("rowGetArrayValue"),
+            sig: JNIString::from("(JILjava/lang/Class;)Ljava/lang/Object;"),
+            fn_ptr: risingwave_java_binding::Java_com_risingwave_java_binding_Binding_rowGetArrayValue as *mut c_void,
+        },
+
+
+        NativeMethod {
+            name: JNIString::from("rowClose"),
+            sig: JNIString::from("(J)V"),
+            fn_ptr: risingwave_java_binding::Java_com_risingwave_java_binding_Binding_rowClose as *mut c_void,
+        },
+
+
+        NativeMethod {
+            name: JNIString::from("streamChunkIteratorNew"),
+            sig: JNIString::from("([B)J"),
+            fn_ptr: risingwave_java_binding::Java_com_risingwave_java_binding_Binding_streamChunkIteratorNew as *mut c_void,
+        },
+
+
+        NativeMethod {
+            name: JNIString::from("streamChunkIteratorNext"),
+            sig: JNIString::from("(J)J"),
+            fn_ptr: risingwave_java_binding::Java_com_risingwave_java_binding_Binding_streamChunkIteratorNext as *mut c_void,
+        },
+
+
+        NativeMethod {
+            name: JNIString::from("streamChunkIteratorClose"),
+            sig: JNIString::from("(J)V"),
+            fn_ptr: risingwave_java_binding::Java_com_risingwave_java_binding_Binding_streamChunkIteratorClose as *mut c_void,
```

Comment:
There are 2 ways to register a native method to the JVM. One is to compile a dynamic library containing the native method's implementation and load the dynamic library on the Java side. Another way is registering just like the above code, i.e. register it at runtime to the JVM directly. I preferred the latter one since I encountered some tricky bugs (static variables usage is prone to error since compute node source codes and the dynamic library source codes are compiled at different times. It is highly possible we allocate some memory in a global allocator A and then deallocate it in another global allocator B) in development when I used the first method.